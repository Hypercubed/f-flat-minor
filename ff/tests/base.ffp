.import ../lib/core.ff
.import ../lib/testing.ffp

0 'Shuffling' prints 
    3 dup 3 assert_eq 3 assert_eq
OK

0 'Basic\sMath' prints 
    1 2 + 3 assert_eq
    3 4 - -1 assert_eq
    5 6 * 30 assert_eq
    8 4 / 2 assert_eq
    8 5 % 3 assert_eq
    9 12 ^ 282429536481 assert_eq
OK


0 'Large\sPowers' prints
    5 4 3 2 ^ ^ ^ 10 10 ^ % 8212890625 assert_eq
OK

0 'Bitwise' prints 
    5 3 & 1 assert_eq
    5 3 | 7 assert_eq
    5 ~ -6 assert_eq
OK

0 'Comparison' prints 
    1 2 < assert
    2 2 < assert_false
    4 3 < assert_false
    5 6 > assert_false
    6 6 > assert_false
    8 7 > assert
    9 10 = assert_false
    10 10 = assert
OK

0 'Control\sFlow' prints 
    5 true [ 3 + ] ? 8 assert_eq
    5 false [ 3 + ] ? 5 assert_eq
OK