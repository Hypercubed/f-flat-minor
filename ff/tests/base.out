Shuffling OK
Basic Math OK
Large Powers OK
Bitwise OK
Comparison OK
Control Flow OK
